src/js/components/Ballot/PositionRowListCompressed.jsx

```diff
@@ -88,9 +90,75 @@ class PositionRowListCompressed extends Component {
     if (this.positionItemTimer) clearTimeout(this.positionItemTimer);
   }
 
-  onBallotStoreChange () {
-    // console.log('PositionRowListCompressed onBallotStoreChange');
-    this.onCachedPositionsChange();
+  handleTalkingAboutClick = () => {
+    const { location: { pathname: currentPathname } } = window;
+    const currentPage = lookupPageNameAndPageTypeDict(currentPathname);
+    const { ballotItemWeVoteId } = this.props;
+
+    const dataLayerObject = {
+      event: 'click_talking_about',
+      destinationDetails: {
+        destinationPageName: currentPage.pageName,
+        destinationPageType: 'CandidateEndorsementModal',
+        destinationPathname: currentPathname,
+      },
+      endorsementDetails: {
+        endorsementCount: this.getEndorsementCount(),
+        interactionType: 'click_talking_about',
+        organizationCount: this.getOrganizationCount(),
+        talkingAboutText: this.getTalkingAboutText(),
+      },
+      pageDetails: {
+        pageName: currentPage.pageName,
+        pageType: currentPage.pageType,
+        pathname: currentPathname,
+      },
+      userDetails: {
+        stateCode: VoterStore.getVoterStateCode(),
+        userCohort: VoterStore.getAnalyticsUserCohort?.() || 'unknown',
+        voterWeVoteId: VoterStore.getVoterWeVoteId(),
+      },
+    };
+    // Add candidate or measure details based on ballotItemWeVoteId
+    if (ballotItemWeVoteId.includes('cand')) {
+      const candidate = CandidateStore.getCandidateByWeVoteId(ballotItemWeVoteId);
+      dataLayerObject.candidateDetails = {
+        candidateWeVoteId: ballotItemWeVoteId,
+        candidateName: candidate ? candidate.ballot_item_display_name : '',
```

Comment:
For getting the candidateName, please use:
```
candidateName: candidate ? CandidateStore.getCandidateName(ballotItemWeVoteId) : '',
```
For the other candidate properties, I would recommend adding a console.log so you can verify the field names, and what the data looks like as you are testing.